File: python/oak/vio_jsonl.py
import depthai
import spectacularAI
import time

pipeline = depthai.Pipeline()

vio_pipeline = spectacularAI.depthai.Pipeline(pipeline)
vio_pipeline.imuToCameraLeft = [
    [0, 1, 0, 0],
    [1, 0, 0, 0],
    [0, 0,-1, 0],
    [0, 0, 0, 1]
]
# optional config args: vio_pipeline = spectacularAI.depthai.Pipeline(pipeline, config, useStereo=False)

with depthai.Device(pipeline) as device, \
    vio_pipeline.startSession(device) as vio_session:

    while True:
        out = vio_session.waitForOutput()
        print(out.asJson())

File: cpp/oak/vio_jsonl.cpp
#include <iostream>
#include <depthai/depthai.hpp>
#include <spectacularAI/depthai/plugin.hpp>
#ifdef EXAMPLE_USE_OPENCV
#include <opencv2/opencv.hpp>
#endif

int main() {
    // Create Depth AI (OAK-D) pipeline
    dai::Pipeline pipeline;

    // Optional configuration
    spectacularAI::daiPlugin::Configuration config;
    // config.useStereo = false;
    // config.useFeatureTracker = false;

    spectacularAI::daiPlugin::Pipeline vioPipeline(pipeline, config);

#ifdef EXAMPLE_USE_OPENCV
    // note: must set useFeatureTracker = false or mono cam will not be read
    vioPipeline.hooks.monoPrimary = [&](std::shared_ptr<dai::ImgFrame> img) {
        // Note: typically not main thread
        cv::imshow("primary mono cam", img->getCvFrame());
        cv::waitKey(1);
    };
#endif

    // Connect to device and start pipeline
    dai::Device device(pipeline);
    auto vioSession = vioPipeline.startSession(device);

    while (true) {
        auto vioOut = vioSession->waitForOutput();
        std::cout << vioOut->asJson() << std::endl;
    }

    return 0;
}
